api4/user_test.go

```diff
@@ -123,25 +151,34 @@ func TestCreateUserInputFilter(t *testing.T) {
 		})
 
 		t.Run("InvalidRole", func(t *testing.T) {
-			user := &model.User{Email: "foobar+testinvalidrole@mattermost.com", Password: "Password1", Username: GenerateTestUsername(), Roles: "system_user system_admin"}
-			_, resp := th.SystemAdminClient.CreateUser(user)
-			CheckNoError(t, resp)
-			ruser, err := th.App.GetUserByEmail("foobar+testinvalidrole@mattermost.com")
-			assert.Nil(t, err)
-			assert.NotEqual(t, ruser.Roles, "system_user system_admin")
+			t.Run("SystemAdminClient", func(t *testing.T) {
+				user := &model.User{Email: "foobar+testinvalidrole@mattermost.com", Password: "Password1", Username: GenerateTestUsername(), Roles: "system_user system_admin"}
+				_, resp := th.SystemAdminClient.CreateUser(user)
+				CheckNoError(t, resp)
+				ruser, err := th.App.GetUserByEmail("foobar+testinvalidrole@mattermost.com")
+				assert.Nil(t, err)
+				assert.NotEqual(t, ruser.Roles, "system_user system_admin")
+			})
+			t.Run("LocalClient", func(t *testing.T) {
+				user := &model.User{Email: "foobar+testinvalidrolelocalclient@mattermost.com", Password: "Password1", Username: GenerateTestUsername(), Roles: "system_user system_admin"}
+				_, resp := th.LocalClient.CreateUser(user)
+				CheckNoError(t, resp)
+				ruser, err := th.App.GetUserByEmail("foobar+testinvalidrolelocalclient@mattermost.com")
+				assert.Nil(t, err)
+				assert.NotEqual(t, ruser.Roles, "system_user system_admin")
+			})
```

Comment:
We can narrow down this one, also we should use `require` instead of `aasert` to avoid panics if the next check is looking for a specific value :)
```golang
	th.TestForSystemAdminAndLocal(t, func(t *testing.T, client *model.Client4) {
		emailAddr := strconv.Itoa(rand.Intn(1000)) + "+testinvalidrole@mattermost.com"
		user := &model.User{Email: emailAddr, Password: "Password1", Username: GenerateTestUsername(), Roles: "system_user system_admin"}
		_, resp := client.CreateUser(user)
		CheckNoError(t, resp)
		ruser, err := th.App.GetUserByEmail(emailAddr)
		require.Nil(t, err)
		assert.NotEqual(t, ruser.Roles, "system_user system_admin")
	}, "InvalidRole")
```